project/src/test/java/com/google/sps/servlets/QueryServletTest.java

```diff
@@ -91,39 +92,89 @@ public void getRequest_fetchMoreThanOneCuisine_success() throws Exception {
     assertEquals(getPlacesAmountInResponse(), QueryServlet.MAX_NUM_PLACES_TO_RECOMMEND);
   }
 
+  @Test
+  public void getRequest_fetchedPlacesShouldBeFiltered_filterPlaces() throws Exception {
+    when(REQUEST.getParameter("rating")).thenReturn("4");
+    Place validPlace = createValidPlaceBuilder().setName("validPlace").build();
+    Place lowRating = createValidPlaceBuilder().setRating(3).setName("lowRatingPlace").build();
+    Place noWebsite = createValidPlaceBuilder().setWebsiteUrl("").setName("noWebsitePlace").build();
+    // The following place purposely has the same name as the valid place, so that we make sure that
+    // we end up with only one of them.
+    Place anotherBranchOfValidPlace = createValidPlaceBuilder().setName("validPlace").build();
```

Comment:
Suggestion: maybe rename to validPlace2?

Reply:
done